detoks/src/main/java/nl/tudelft/trustchain/detoks/TorrentManager.kt

```diff
@@ -255,19 +259,24 @@ class TorrentManager private constructor (
         for (it in 0 until torrentInfo.numFiles()) {
             val fileName = torrentInfo.files().fileName(it)
             if (fileName.endsWith(".mp4")) {
-                torrentFiles.add(
-                    TorrentHandler(
-                        cacheDir,
-                        handle,
-                        torrentInfo.name(),
-                        fileName,
-                        it
-                    )
+                val torrent = TorrentHandler(
+                    cacheDir,
+                    handle,
+                    torrentInfo.name(),
+                    fileName,
+                    it
                 )
+                torrentFiles.add(torrent)
+                addProfile(torrent.handle.makeMagnetUri())
             }
         }
     }
 
+    fun addProfile(magnet: String) {
+        profile.profiles[magnet] = ProfileEntry(
+            uploadDate = (getInfoFromMagnet(magnet)?:return).creationDate())
+    }
+
```

Comment:
I think this function should be in the profiles class, since it has nothing to do with the TorrentManager. Also, since everywhere else you now make a ProfileEntry I don't think the uploadDate is actually initialised (like in updateWatchTime when there isn't an entry yet). So it would be better to move to the Profile class and then consistently use it everywhere. 
Maybe uploadDate could also be initialised in the constructor? then it is at least always initialised even when this method isn't called

Reply:
Moved the function but it needs the torrentinfo so I take it as parameter now, added a function generateUploadDate as I see no way to get the torrent info into the constructor of the profile entry without changing a significant amount of code